modules/siddhi-extensions/output-transports/jms-output-transport/src/test/java/org/wso2/siddhi/extension/output/transport/jms/util/JMSClient.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.siddhi.extension.output.transport.jms.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import javax.jms.QueueConnectionFactory;
+import javax.jms.TopicConnectionFactory;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import java.io.IOException;
+import java.util.Properties;
+
+public class JMSClient implements Runnable{
+    private static Log log = LogFactory.getLog(JMSClient.class);
+
+    private String broker, topic, queue;
+
+    public JMSClient(String broker, String topic, String queue) {
+        this.broker = broker;
+        this.topic = topic;
+        this.queue = queue;
+    }
+
+    public void listen() throws InterruptedException {
+        Properties properties = new Properties();
+        try {
+            boolean validBroker = true;
+            if (broker.equalsIgnoreCase("qpid")) {
+                properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("qpid.properties"));
+            } else if (broker.equalsIgnoreCase("activemq")) {
+                properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("activemq.properties"));
+            } else if (broker.equalsIgnoreCase("mb")) {
+                properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("mb.properties"));
+            } else {
+                validBroker = false;
+                log.error("Entered broker is invalid! ");
+            }
+            if (validBroker) {
+                if (topic != null && topic.isEmpty() || topic.equals("\"\"")) {
+                    topic = null;
+                }
+                if (queue != null && queue.isEmpty() || queue.equals("\"\"")) {
+                    queue = null;
+                }
+                if (topic == null && queue == null) {
+                    log.error("Enter topic value or queue value! ");
+                } else if (topic != null) {
```

Comment:
if both queue and topic given we give priority to topic by default and ignore queue??